main.go

```diff
@@ -83,6 +85,18 @@ func main() {
 		os.Exit(1)
 	}
 
+	k8sconfig, err := rest.InClusterConfig()
```

Comment:
I don't see a need for this. I see that we rely on this for deriving the service IP of metadata cache service and instead we can just resolve the metadata service name to get the service IP..